src/knex.js

```diff
@@ -0,0 +1,83 @@
+import Raw from './raw';
+import Client from './client';
+
+import makeKnex from './util/make-knex';
+import parseConnection from './util/parse-connection';
+import { SUPPORTED_CLIENTS, CLIENT_ALIASES } from './constants';
+
+export default function Knex(config) {
+  // If config is string, try to parse it
+  if (typeof config === 'string') {
+    const parsedConfig = Object.assign(parseConnection(config), arguments[2]);
+    return new Knex(parsedConfig);
+  }
+
+  let Dialect;
+  // If user provided no relevant parameters, use generic client
+  if (arguments.length === 0 || (!config.client && !config.dialect)) {
+    Dialect = Client;
+  }
+
+  // If user provided Client constructor as a parameter, use it
+  else if (
+    typeof config.client === 'function' &&
+    config.client.prototype instanceof Client
+  ) {
+    Dialect = config.client;
+  }
+
+  // If neither applies, let's assume user specified name of a client or dialect as a string
+  else {
+    const clientName = config.client || config.dialect;
+    if (!SUPPORTED_CLIENTS.includes(clientName)) {
+      throw new Error(
+        `knex: Unknown configuration option 'client' value ${clientName}. Note that it is case-sensitive, check documentation for supported values.`
+      );
+    }
+
+    Dialect = require(`./dialects/${CLIENT_ALIASES[clientName] ||
+      clientName}/index.js`);
+  }
+
+  // If config connection parameter is passed as string, try to parse it
+  if (typeof config.connection === 'string') {
+    config = Object.assign({}, config, {
+      connection: parseConnection(config.connection).connection,
+    });
+  }
+  return makeKnex(new Dialect(config));
+}
+
+// Expose Client on the main Knex namespace.
+Knex.Client = Client;
+
+/* eslint no-console:0 */
+
+Object.defineProperties(Knex, {
+  VERSION: {
+    get() {
+      console.warn(
+        'Knex.VERSION is deprecated, you can get the module version' +
+          "by running require('knex/package').version"
+      );
+      return '0.12.6';
```

Comment:
Looks like this return value hasn't been updated in some time :laughing: Maybe it's time to remove this deprecation warning? I know it's not related to the rest of the work done here, so feel free to ignore this comment.

Reply:
So you suggest to remove entire
```
VERSION: {
    get() {
      console.warn(
        'Knex.VERSION is deprecated, you can get the module version' +
          "by running require('knex/package').version"
      );
      return '0.12.6';
}
```

block? Sure, I can do that.

Reply:
Yes. The message has been there for almost 2 years now. I think that's enough time for a deprecation warning.